cmd/util/ledger/migrations/static_type_migration.go

```diff
@@ -8,18 +8,22 @@ import (
 type StaticTypeMigrationRules map[common.TypeID]interpreter.StaticType
 
 func NewStaticTypeMigration[T interpreter.StaticType](
-	rules StaticTypeMigrationRules,
+	rulesGetter func() StaticTypeMigrationRules,
```

Comment:
Is there a particular reason why this is a getter, and the rules are not passed as-is?

Reply:
This constructor is called at the time of the constructing the migrations (i.e: Cadence value migration). But the rules can only be finalized after running this newly added migration. i.e: the passed `LegacyTypeRequirements` list is lazily loaded. So we need to delay the construction of the rules, until after the execution of previous 
 migration (`TypeRequirementsExtractingMigration` to be exact).

Reply:
I see! Can you please leave some comments like this explanation in the code?